src/EditorFeatures/Core/Implementation/Preview/PreviewFactoryService.cs

```diff
@@ -564,24 +566,27 @@ private ITextBuffer CreateNewPlainTextBuffer(TextDocument document, Cancellation
             Contract.ThrowIfTrue((leftWorkspace == null) && (rightWorkspace == null));
 
             var diffBuffer = _differenceBufferService.CreateDifferenceBuffer(
-                    originalBuffer, changedBuffer,
-                    new StringDifferenceOptions(), disableEditing: true);
+                originalBuffer, changedBuffer,
+                new StringDifferenceOptions(), disableEditing: true);
 
             var diffViewer = _differenceViewerService.CreateDifferenceView(diffBuffer, _previewRoleSet);
 
             diffViewer.Closed += (s, e) =>
             {
-                if (leftWorkspace != null)
-                {
-                    leftWorkspace.Dispose();
-                    leftWorkspace = null;
-                }
-
-                if (rightWorkspace != null)
-                {
-                    rightWorkspace.Dispose();
-                    rightWorkspace = null;
-                }
+                // Workaround Editor bug.  The editor has an issue where they sometimes crash when 
+                // trying to apply changes to projection buffer.  So, when the user actually invokes
+                // a SuggestedAction we may then edit a text buffer, which the editor will then 
+                // try to propagate through the projections we have here over that buffer.  To ensure
+                // that that doesn't happen, we clear out the projections first so that this crash
+                // won't happen.
+                originalBuffer.DeleteSpans(0, originalBuffer.CurrentSnapshot.SpanCount);
+                changedBuffer.DeleteSpans(0, changedBuffer.CurrentSnapshot.SpanCount);
+
+                leftWorkspace?.Dispose();
+                leftWorkspace = null;
```

Comment:
I believe @heejaechang did a hotfix for the fact that we were leaking workspaces from previews.